mypy/semanal.py

```diff
@@ -2288,6 +2288,9 @@ def analyze_member_lvalue(self, lval: MemberExpr, explicit_type: bool, is_final:
             if cur_node and is_final:
                 # Overrides will be checked in type checker.
                 self.fail("Cannot redefine an existing name as final", lval)
+            if (cur_node and isinstance(cur_node.node, Var) and cur_node.node.is_inferred and
+                    explicit_type):
+                self.name_already_defined('self.' + lval.name, lval, cur_node)
```

Comment:
I wouldn't use this error message because `self.foo` is not a name, it is an attribute. I would better use `Attribute "foo" already defined ...`.

Reply:
Fixed.